From 321f1ee8907961873f4caccd6d25d2a78ef4ae85 Mon Sep 17 00:00:00 2001
From: "qiyao.zhou" <qiyao.zhou@amlogic.com>
Date: Fri, 23 Apr 2021 17:35:12 +0800
Subject: [PATCH] frameworks: add static frame setting flow. [2/2]

PD#SWPL-44606

Problem:
1.need add blue screen function.
2.need add static frame function for channal.

Solution:
1.add this feature to droidtvsetting.
2.add interface in systemcontrol.

Verify:
ok on redi.

Change-Id: I02287652a9758e0081d3062c9a480bba4b13e0f5
---
 media/java/android/media/tv/TvView.java | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/media/java/android/media/tv/TvView.java b/media/java/android/media/tv/TvView.java
index 6b329f8ec694..38b54c45911a 100644
--- a/media/java/android/media/tv/TvView.java
+++ b/media/java/android/media/tv/TvView.java
@@ -53,7 +53,7 @@ import java.lang.ref.WeakReference;
 import java.util.ArrayDeque;
 import java.util.List;
 import java.util.Queue;
-
+import android.os.SystemProperties;
 /**
  * Displays TV contents. The TvView class provides a high level interface for applications to show
  * TV programs from various TV sources that implement {@link TvInputService}. (Note that the list of
@@ -125,6 +125,7 @@ public class TvView extends ViewGroup {
         @Override
         public void surfaceCreated(SurfaceHolder holder) {
             mSurface = holder.getSurface();
+            mSurfaceView.setResizeBackgroundColor(isBlueScreenEnabled() ? 0xff0000ff : 0);
             setSessionSurface(mSurface);
         }
 
@@ -366,7 +367,11 @@ public class TvView extends ViewGroup {
             mUseRequestedSurfaceLayout = false;
             mSession.release();
             mSession = null;
-            resetSurfaceView();
+            if (mSurface == null) {
+                resetSurfaceView();
+            } else {
+                mSurfaceView.setResizeBackgroundColor(isBlueScreenEnabled() ? 0xff0000ff : 0);
+            }
         }
     }
 
@@ -772,6 +777,11 @@ public class TvView extends ViewGroup {
         }
     }
 
+    private boolean isBlueScreenEnabled () {
+        //no need show blue screen on preview window
+        return getWidth() > 720 && getHeight() > 480 && SystemProperties.getBoolean("persist.tv.blue.screen.enabled", false);
+    }
+
     private void resetSurfaceView() {
         if (mSurfaceView != null) {
             mSurfaceView.getHolder().removeCallback(mSurfaceHolderCallback);
-- 
2.29.0

